specs/CONFIG.md

```diff
@@ -0,0 +1,207 @@
+# libp2p Daemon Config Spec
+
+The libp2p daemon is an interface to libp2p, and as such has a large number of 
+configuration options. For complex deployments, these options can become 
+unwieldy on the command line. For this, we allow the ability to provide a
+configuration spec via json from a file or from stdin
+
+_At the moment, this is a living document. As such, it will be susceptible to
+changes until stabilization._
+
+## Considerations
+
+As this is a spec that must be shared between multiple implementations, it is 
+probably best that the structure be generated from some shared source of truth,
+instead of depending on discipline to keep the implementation consistent 
+between language implementations. For instance, the config structure could be 
+generated as a protobuf message.
+
+## Command Line
+
+There are two ways to provide a JSON configuration to the daemon. Both methods 
+cause all other configuration command line options to be ignored.
+
+### Read from a file
+`p2pd -f ./conf.json`
+
+### Read from stdin
+`cat ./conf.json | p2pd -i`
+
+
+## Schema
+
+* `Field Name`
+    * Description
+    * Type
+    * `default`
+
+* `Listen`
+    * Daemon control listen multiaddr
+    * Maddr String
+    * `"/unix/tmp/p2pd.sock"`
+* `Quiet`
+    * Be Quiet
+    * Boolean
+    * `false`
+* `ID`
+    * Peer identity; private key file
+    * String
+    * `""`
+* `Bootstrap`
+    * `Enabled`
+        * Connects to bootstrap peers and bootstraps the dht if enabled
+        * Boolean
+        * `false`
+    * `Peers`
+        * List of bootstrap peers; defaults to the IPFS DHT peers
+        * Array[Maddr String]
+        * `[]`
+* `DHT`
+    * `Enabled`
```

Comment:
I would rather call this `Server` and `Client` than enabled, as `ClientMode` and `Enabled`. ClientMode also enabls the dht.

Reply:
Unless there is an implication that we need to set both `Enabled` and `ClientMode` to get the dht client behaviour.

Reply:
At least in the Go implementation, the don't seem to depend on each other. Preferably it would be one value that would be chosen from an Enum, though, unless it is possible to have both enabled at the same time?